src/ngraph/node.hpp

```diff
@@ -430,6 +436,7 @@ namespace ngraph
         std::unordered_map<Node*, autodiff::Adjoints> m_adjoint_map;
         Placement m_placement = Placement::DEFAULT;
         size_t m_placement_index = placement_invalid;
+        size_t m_opset_version = 0;
```

Comment:
So is this the _earliest_, or the _latest_, opset version that this class belongs to? In the end I think this is the only issue I have that doesn't fall into the bucket of "implementation details". At a minimum, we probably need a way to represent that "this class represents `Reshape` as of opset versions `v3, v4, v5`".